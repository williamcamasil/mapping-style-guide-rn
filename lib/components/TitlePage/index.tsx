import React from 'react';
import { StyleSheet, View } from 'react-native';

import { SvgProps } from 'react-native-svg';

import { useViewStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import Spacer from '../Spacer';
import Text from '../Text';

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
  },
  fill: {
    flex: 1,
  },
});

export type TitlePagePropsType = {
  theme: AppThemeType;
  title: string;
  description?: string;
  Logo?: React.FunctionComponent<SvgProps>;
};

const TitlePage = ({
  theme,
  title,
  description,
  Logo,
}: TitlePagePropsType) => {
  const containerStyles = useViewStyles(() => [
    styles.row,
    {
      columnGap: theme.spacings.sNano,
    },
  ], [theme.spacings.sNano]);

  const renderDescription = () => {
    if (!description) {
      return null;
    }
    return (
      <>
        <Spacer size={theme.spacings.sXXS} />
        <Text>{description}</Text>
      </>
    );
  };

  const renderLogo = () => {
    if (!Logo) return null;
    return (
      <View>
        <Logo />
      </View>
    );
  };

  return (
    <View style={containerStyles}>
      <View style={styles.fill}>
        <Text variant="headingSmall" color="neutralGray700">
          {title}
        </Text>
        {renderDescription()}
      </View>
      {renderLogo()}
    </View>
  );
};

export default withTheme(TitlePage);
