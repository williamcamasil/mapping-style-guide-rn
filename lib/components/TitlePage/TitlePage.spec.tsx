import React from 'react';

import { render } from '@testing-library/react-native';

import TitlePage from '.';

describe('TitlePage', () => {
  it('Should match snapshot with description', () => {
    const component = render(<TitlePage title="Title" description="Description" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should match snapshot with title only', () => {
    const component = render(<TitlePage title="Title" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
