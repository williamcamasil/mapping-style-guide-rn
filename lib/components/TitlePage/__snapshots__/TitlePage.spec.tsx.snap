// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitlePage Should match snapshot with description 1`] = `
<View
  style={
    {
      "columnGap": 8,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 23,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 28,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#676D79",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Description
    </Text>
  </View>
</View>
`;

exports[`TitlePage Should match snapshot with title only 1`] = `
<View
  style={
    {
      "columnGap": 8,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 23,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 28,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
